packages/core/src/storage/base.ts

```diff
@@ -18,7 +18,7 @@ export abstract class MastraStorage extends MastraBase {
   /** @deprecated import { TABLE_TRACES } from '@mastra/core/storage' instead */
   static readonly TABLE_TRACES = TABLE_TRACES;
 
-  hasInit = false;
+  protected hasInitialized: boolean | Promise<any> = false;
```

Comment:
You probably want to do call it more like this
```suggestion
  protected hasInitializedPromise: null | Promise<boolean> = null;
```

Reply:
Oops, yes I do, thanks